api/client/commands.go

```diff
@@ -373,6 +374,11 @@ func (cli *DockerCli) CmdLogin(args ...string) error {
 			email = readInput(cli.in, cli.out)
 			if email == "" {
 				email = authconfig.Email
+			} else {
+				reg := regexp.MustCompile(`^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$`)
```

Comment:
I think more is needed in this regex - I'd check the RFC ( http://tools.ietf.org/html/rfc822#section-6.1 )